public/pages/Main/Main.tsx

```diff
@@ -151,28 +159,42 @@ const dataSourceEnabledPaths: string[] = [
   ROUTES.COMPOSABLE_TEMPLATES,
   ROUTES.CREATE_COMPOSABLE_TEMPLATE,
   ROUTES.REINDEX,
+  ROUTES.ROLLUPS,
+  ROUTES.CREATE_ROLLUP,
+  ROUTES.EDIT_ROLLUP,
+  ROUTES.ROLLUP_DETAILS,
+  ROUTES.TRANSFORMS,
+  ROUTES.CREATE_TRANSFORM,
+  ROUTES.EDIT_TRANSFORM,
+  ROUTES.TRANSFORM_DETAILS,
 ];
 
 export default class Main extends Component<MainProps, MainState> {
   constructor(props: MainProps) {
     super(props);
+    let dataSourceId = "";
+    let dataSourceLabel = "";
     if (props.multiDataSourceEnabled) {
-      const { dataSourceId = "", dataSourceLabel = "" } = queryString.parse(this.props.location.search) as {
+      const { dataSourceId: parsedDataSourceId, dataSourceLabel: parsedDataSourceLabel } = queryString.parse(
+        this.props.location.search
+      ) as {
         dataSourceId: string;
         dataSourceLabel: string;
       };
-      this.state = {
-        dataSourceId: dataSourceId,
-        dataSourceLabel: dataSourceLabel,
-      };
-    } else {
-      this.state = {
-        dataSourceId: "",
-        dataSourceLabel: "",
-      };
+      dataSourceId = parsedDataSourceId || "";
+      dataSourceLabel = parsedDataSourceLabel || "";
     }
+    this.state = {
+      dataSourceId: dataSourceId,
+      dataSourceLabel: dataSourceLabel,
+      dataSourceReadOnly: false,
```

Comment:
Can we honor the readonly from the data source picker component? I think a single source of this state should be good enough.

Reply:
For pages like rollup and transform which are multi-step forms with same URL we need the individual components to control the state of data source picker by making it writable and read only as required. This invalidates our existing approach of mounting the data source picker component based on URL. Therefore, we needed a state for child components to change and control the data source picker

Reply:
I see, thanks for the clarification.